src/components/Filter/Filter.tsx

```diff
@@ -52,6 +63,43 @@ const Filter: React.FC<Props & TestProps> = props => {
     onSelect(option);
   }
 
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
@panvourtsis is this supposed to have the exact same functionality as async select component?

If yes, the we could abstract this logic too a separate utils file and reuse it in both components.

Reply:
Inside `Select` component we have used `onInput` and not `onChange` but yes the implementation is almost the same. I would rather not to touch other components in this MR. How about to open a ticket to refactor these parts to both components in a separated PR?

Reply:
The type of the change event could be abstracted as a generic type and be passed by each component so it could satisfy both cases. 
In general I prefer things to be changed on the spot and not add tech debt but I don't mind this happening in a separate branch as long as it is documented and changed.

Reply:
ok yes i agree for the type of change event and i made the change to both files. thank you!

regarding the the re-usable logic, I am happy to do the change, but to be honest i am not sure that we have much to gain. Maybe i miss something but it would a function that should have to take 5 arguments (the isSearchable prop, the setSearchValue function, the event, the minCharactersToSearch prop and the debouncedOnChange function) and its a lot, isn't?

Reply:
It might be a lot but I see more value in keeping all the search logic in one place rather than having to maintain the same logic in two (for now) places.

I would expect all the search functionalities in the app to be the same and work with the same way, something that, as we keep adding new components and the code base gets larger,  it's going to be harder to maintain duplicate code, let alone remember it exists. 

I am not going to block the ticket by insisting even more on this. At least add a TODO in both places that the code is pretty much duplicated and we need to figure out a way of refactoring it. ( Also it will help maintain it as long as it stays like this )

Reply:
@Christos-Zacharopoulos thank you for the nice conversation and suggestion! I think you are right and I made the changes as you said :) 

Thanks again!